staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -975,6 +975,10 @@ func (c *Cacher) GuaranteedUpdate(
 
 // Count implements storage.Interface.
 func (c *Cacher) Count(pathPrefix string) (int64, error) {
+	if c.ready.check() {
+		cnt := c.watchCache.Count()
+		return cnt, nil
+	}
 	return c.storage.Count(pathPrefix)
```

Comment:
I have a little concern here, maybe it's insignificant.

In `Cacher`, the semantic of `Count(pathPrefix string) (int64, error)` is to filter based on pathPrefix and return the count of objects that match the prefix. But in this implement, the prefix is ignored. 

The prefix in an argument here because just for one kind of resource, the prefix can also be different across different namespace. I know this method is only used for metrics, and there is no such caller will pass a namespaced prefix for now, but ignoring the prefix seems break the pervious semantic and easy to leads problem in future. 

IMO, there are two ways to address this:
1. we can list all key and filter by the prefix and then return the count. this could be very Inefficiency as discussed already, so we nearly willn't to do this.
2. we can add a check to the prefix. if the prefix is for all-namespaces and then we can use the cache, otherwise fallback to etcd or listing and filter all keys.